ui/components/ErrorBoundary.js

```diff
@@ -1,41 +1,28 @@
-import { Button } from '@material-ui/core';
 import React from 'react';
+import { Button, ErrorBoundary as SistentErrorBoundary } from '@layer5/sistent';
 
 /**
  * ErrorBoundary is a React component that catches JavaScript errors in its child components and renders a fallback UI when an error occurs.
  * It should be used as a wrapper around components that might throw errors.
- * @deprecated use error boundary from sistent instead
  */
-class ErrorBoundary extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { hasError: false, error: null };
-  }
 
-  /** Update state so the next render will show the fallback UI. */
-  static getDerivedStateFromError(error) {
-    return { hasError: true, error: error };
-  }
-
-  resetErrorBoundary = () => {
-    this.setState({ hasError: false, error: null });
-  };
-
-  /** You can render any custom fallback UI */
-  render() {
-    if (this.state.hasError) {
-      return (
+function ErrorBoundary({ children }) {
+  return (
+    <SistentErrorBoundary
+      FallbackComponent={({ error, resetErrorBoundary }) => (
         <div className="alert alert-danger">
           <p>Couldn&apos;t open form. Encountered the following error:</p>
-          <pre>{this.state.error.message}</pre>
-          <Button color="primary" variant="contained" onClick={this.resetErrorBoundary}>
+          <pre>{error.message}</pre>
+          <Button color="primary" variant="contained" onClick={resetErrorBoundary}>
             Refresh Form
           </Button>
         </div>
-      );
-    }
-    return this.props.children;
-  }
+      )}
+      onError={(error) => console.error('Error in Spaces Preferences Component', error)}
```

Comment:
hey thanks for the changes 🚀 . Curious if we could break out the definition of the fallbackComponent and the error handling just in case we need to add additional logic in the future to these? 

So return would like something like 
```
  return (
    <SistentErrorBoundary 
      FallbackComponent={FallbackComponent}
      onError={handleError}
    >
      {children}
    </SistentErrorBoundary>
  );
```

Reply:
Done .